### PR TITLE
Add a separate lesson on string templates: f-strings and `format`

### PR DESCRIPTION
**Pull request by encukou**
Here's my explanation on f-strings, improved over the last 3 "experimental" courses in Brno. I'm now pretty happy with it.
Also, I moved the text on `format` after it (with common explanations removed), because f-strings don't cover all the use cases.

Format specifications like `{number:08x}` and details like `'a[k]'.format(a)` are still left out.

Sadly, this didn't work-out time-wise wrt. https://github.com/PyLadiesCZ/naucse.python.cz/pull/48

### CODE REVIEW

File: lessons/beginners/fstring/info.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+title: Šablony (formátovací řetězce)
+style: md
+attribution:
+- Pro PyLadies Brno napsal Petr Viktorin, 2018-2019.
+license: cc-by-sa-40
```
Comment (hroncok) [inline lessons/beginners/fstring/info.yml]: Is it possible to use CC0 for new code snippets?

```suggestion
license: cc-by-sa-40
license_code: cc0
```

Comment (encukou) [reply lessons/beginners/fstring/info.yml]: Yes. I should go through the old ones, too.

File: lessons/beginners/fstring/index.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+S pozdravem,
+{podpis}.
+```
+
+Takovou šablonu můžeš použít jako *formátovací řetězec*
+(angl. [*formatted string literal*](https://docs.python.org/3.6/reference/lexical_analysis.html#formatted-string-literals),
+zkráceně *f-string*).
+Jako jakýkoli jiný řetězec ji vlož do uvozovek.
+A aby bylo jasné že jde o šablonu, před první uvozovky dej značku `f`.
```
Comment (janpipek) [inline lessons/beginners/fstring/index.md]:
```suggestion
A aby bylo jasné, že jde o šablonu, před první uvozovky přidej navíc značku `f`.
```

File: lessons/beginners/fstring/index.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+pojmenuješ – tedy uložíš do vhodně pojmenované proměnné.
+
+
+## Metoda format
+
+Někdy se stane, že jednu šablonu potřebuješ použít vícekrát.
+Pak formátovací řetězec použít nemůžeš, protože se do něj proměnné doplňují
+automaticky a hned.
+Pro takové případy existuje metoda `format`:
+
```
Comment (janpipek) [inline lessons/beginners/fstring/index.md]: I would add some explicit notion about the "f" not being present here.

File: lessons/beginners/fstring/index.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
+# Šablony (formátovací řetězce)
+
+Řekněme, že chceš vypsat určitou hodnotu uživatelovi s nějakou „omáčkou“ okolo.
```
[the remark this answers is not in the document]
Comment (frenzymadness) [reply lessons/beginners/fstring/index.md]: +1

**Comment by frenzymadness** [comment]
I like it. Good job.

**Comment by encukou** [comment]
Thanks for the reviews!

**Pull request merged**